types/utils.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import "encoding/json"
+
+// SortedJSON takes any JSON and returns it sorted by keys. Also, all white-spaces
+// are removed.
+// This method can be used to canonicalize JSON to be returned by GetSignBytes,
+// e.g. for the ledger integration.
+// If the passed JSON isn't valid it will return an error.
+func SortJSON(toSortJSON []byte) ([]byte, error) {
+	var c interface{}
+	err := json.Unmarshal(toSortJSON, &c)
+	if err != nil {
+		return nil, err
+	}
+	js, err := json.Marshal(c)
+	if err != nil {
+		return nil, err
+	}
+	return js, nil
+}
+
+// MustSortJSON is like SortJSON but panic if an error occurs, e.g., if
+// the passed JSON isn't valid.
+func MustSortJSON(toSortJSON []byte) []byte {
+	js, err := SortJSON(toSortJSON)
+	if err != nil {
+		panic(err)
+	}
+	return js
+}
```

Comment:
This is basically copy & pasted from the SDK (added here: https://github.com/cosmos/cosmos-sdk/commit/b7f902fba5fcb367d82a2dd9e59600e91d2dfc72#diff-da7a1d1eef56dc7672476c992ce1e56c) 

I think we should also copy over the tests then. Probably these methods (and tests) do not need to live in both repositories (but that is up to the SDK team).

Reply:
Should we move them to `libs/common` package?

Reply:
Yes, that is probably a good place for this to live in.

Reply:
I'm super OK with not duplicating code. I didn't want to import the SDK into tendermint tho... How should I move forward here?

Reply:
I think we should move those methods into libs/common together with their tests. Whenever the sdk updates to a Tendermint Version including this PR, they could delete the code from the SDK.

Reply:
https://github.com/tendermint/tendermint/pull/3650